### ISSUE TITLE
Combine unload pathing issues

### ISSUE THREAD

**Issue by Imsvale** [Courseplay/courseplay#7215]
* Courseplay v6.04.0000
* Varitron 470 potato harvester
* A few small/medium tractors with trailers

I'm running a potato harvest contract on Ravenport field 8, which obviously takes a long time and is probably not worth it, but hey I'm doing it anyway. I've run various attempts with 1, 2 and 3 unloader tractors, and they all have this fairly frequent issue of maneuvering too close to the harvester and getting either physically stuck on the harvester, or blocked by it (harvester stops when it's full, won't move out of the way of tractor, and tractor set up a path that took it into the harvester), or indeed blocking the harvester. Sometimes the unloader is coming in from right behind the harvester, not quite lined up (with tons of space for lining up much earlier, just not being used), and starts turning way too close to the harvester. Zero margin for error, and accidents often happen. Other times they're cutting in front, assuming too much about the harvester's path. They then stop in front, blocking the harvester, and everything stops. This has been particularly troublesome when the harvester is changing lanes. Overall they're just getting in each other's way a lot.

When the unloader is full, it tends to drive alongside the harvester and is unable to speed up to go to the unload point, until the harvester turns away somehow. 

A separate issue I'm also having is that the unloaders occasionally "forget" which field/harvester they're associated with, and stop working because of that.

Not speeding up while beside harvester:
![2021-05-22 13_30_34-Greenshot](https://user-images.githubusercontent.com/16666305/119225336-7a889700-bb03-11eb-8061-7f3cf608db7f.png)

Only drove on after the harvester turned. Meanwhile the first unloader was blocking the next unloader, and the harvester got almost full:
![2021-05-22 13_32_06-Greenshot](https://user-images.githubusercontent.com/16666305/119225347-8aa07680-bb03-11eb-92a8-15bbf0f136a3.png)

Unloader coming in from weird angles, not giving itself space to maneuver, and getting itself stuck:
![2021-05-22 13_40_08-Greenshot](https://user-images.githubusercontent.com/16666305/119225411-f97dcf80-bb03-11eb-897f-e668a44008d7.png)

It has also forgotten which field it was associated with (this is Field 8):
![2021-05-22 13_40_17-Greenshot](https://user-images.githubusercontent.com/16666305/119225416-fd115680-bb03-11eb-999f-6236ebe11cb4.png)

I've manually told it to drive on, but it doesn't until its path gets far enough away from the harvester:
![2021-05-22 13_47_55-Greenshot](https://user-images.githubusercontent.com/16666305/119225497-63967480-bb04-11eb-8cfb-df134acdf19f.png)

Complaining about traffic when he's waiting in line behind another unloader for the same field and harvester:
![2021-05-22 13_50_18-Greenshot](https://user-images.githubusercontent.com/16666305/119225575-c6880b80-bb04-11eb-9810-01859fd053cc.png)

Incidentally, the unloader in front has again forgotten what field and harvester he's working for:
![2021-05-22 13_50_28-Greenshot](https://user-images.githubusercontent.com/16666305/119225611-f20af600-bb04-11eb-91d9-7d0d6c1b9074.png)

I set the field and harvester, and the guy in front drove off (because he was the closest, he would go next). Guy coming in behind after unloading has also forgotten his field and harvester associations:
![2021-05-22 13_53_41-Greenshot](https://user-images.githubusercontent.com/16666305/119225636-27afdf00-bb05-11eb-899f-1eafaa884fc5.png)

He's also trying to drive through the tractor in front:
![2021-05-22 13_55_09-Greenshot](https://user-images.githubusercontent.com/16666305/119225662-5b8b0480-bb05-11eb-9d99-4e61a151cbfb.png)

Maneuvering in a space-efficient manner is all well and good with respect to trees and structures around the field, but when the vehicle frequently gets into accidents with other courseplay vehicles on the field, you lose ten if not a hundred times the time you would have saved. I often see them turning with absolutely zero margin near other vehicles (as close as they can physically get without their hitboxes impacting). That's just asking for trouble.

**Comment by Tensuko** [comment]
Getting stuck next to combine:
Offset for the unloader, try +0,5. This is the result of an angled field where the unloader is higher so the convayer belt sits lower.

Weird Angle:
That looks like avoid fruit = off to me, turn it on, anything else is a direct path and leads to that problem.

Search Combine on Field:
Ppl still get that wrong.... This is just a sorting/filtering function.
You don't have to choose the Field, you only select the combine! Nothing else will tell the driver to drive to the combine.

**Comment by Imsvale** [comment]
> Getting stuck next to combine:
> Offset for the unloader, try +0,5. This is the result of an angled field where the unloader is higher so the convayer belt sits lower.

The tractor with the bigger trailer already has this. The offset took care of the isolated case where the trailer would get snagged on the belt. It hasn't fixed what I'm talking about in this post.

> Weird Angle:
> That looks like avoid fruit = off to me, turn it on, anything else is a direct path and leads to that problem.

Avoid fruit has been on for all the unloaders, all the time. It doesn't seem to work for potatoes. I considered putting it in the first post, but ended up not doing it, because it was already so large.

> Search Combine on Field:
> Ppl still get that wrong.... This is just a sorting/filtering function.
> You don't have to choose the Field, you only select the combine! Nothing else will tell the driver to drive to the combine.

Right, but the unloaders still forget their combine assignment and get stuck doing nothing.

**Comment by Tensuko** [comment]
logfiel with debugchannel 2 and 11 active.

**Comment by Imsvale** [comment]
The debug channels are disabled for this version, are they not? Is there a way for me to enable this anyway?

**Comment by Tensuko** [comment]
https://github.com/Courseplay/courseplay/releases/tag/6.03.00061
This is basically the same version.

**Comment by Imsvale** [comment]
Alright, I'll try that and post the log file.

**Comment by Tensuko** [comment]
Closed due to inactivity.
